File: exercise/division/evaluation/Evaluator.java
public class Evaluator {

    public static EvaluationResult evaluate(Object actual) {
        if (actual instanceof ArithmeticException) {
            return EvaluationResult.builder(true)
                    .withReadableExpected(actual.toString())
                    .withReadableActual(actual.toString())
                    .build();
        } else {
            return EvaluationResult.builder(false)
                    .withReadableExpected("ArithmeticException")
                    .withReadableActual(actual == null ? "" : actual.toString())
                    .withMessage(new EvaluationResult.Message("Expected ArithmeticException, got something else."))
                    .build();
        }
    }

}

File: exercise/division/evaluation/Evaluator.kt
class Evaluator {
    companion object {
        @JvmStatic
        fun evaluate(actual: Any?): EvaluationResult {
            return if (actual is ArithmeticException) {
                EvaluationResult.Builder(result = true,
                        readableExpected = actual.toString(),
                        readableActual = actual.toString()).build()
            } else {
                EvaluationResult.Builder(result = false,
                        readableExpected = "ArithmeticException",
                        readableActual = actual?.toString() ?: "")
                        .withMessage(EvaluationResult.Message("Expected ArithmeticException, got something else."))
                        .build()
            }
        }
    }
}